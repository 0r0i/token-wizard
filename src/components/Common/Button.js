import React from 'react'

export const Button = ({ buttonText = '', disabled = false, extraClassName = '', onClick, type = 'button' }) => (
  <button className={`sw-Button ${extraClassName}`} disabled={disabled} onClick={onClick} type={type}>
    {buttonText}
  </button>
)
