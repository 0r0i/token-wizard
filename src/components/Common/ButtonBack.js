import React, { Component } from 'react'

export class ButtonBack extends Component {
  /**
   * Render method for ButtonBack component
   * @returns {*}
   */
  render() {
    const { onClick } = this.props

    return (
      <button onClick={onClick} type="button" className={`sw-ButtonBack`}>
        Back
      </button>
    )
  }
}
