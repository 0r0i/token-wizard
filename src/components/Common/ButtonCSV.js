import React from 'react'

export const ButtonCSV = ({ onClick, text, extraClassName = '', disabled }) => (
  <button className={`sw-ButtonCSV ${extraClassName}`} disabled={disabled} onClick={onClick} type="button">
    {text}
  </button>
)
