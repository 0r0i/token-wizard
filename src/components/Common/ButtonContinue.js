import React from 'react'

export const ButtonContinue = ({ disabled = false, type, onClick, extraClassName, buttonText = 'Continue' }) => (
  <button
    className={`sw-ButtonContinue ${extraClassName ? extraClassName : ''}`}
    disabled={disabled}
    onClick={onClick}
    type={type}
  >
    <span className={`sw-ButtonContinue_Text`}>{buttonText}</span>
  </button>
)
