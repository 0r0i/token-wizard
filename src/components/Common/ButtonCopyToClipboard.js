import React from 'react'

export const ButtonCopyToClipboard = ({ disabled, extraClassName, value }) => (
  <button
    className={`sw-ButtonCopyToClipboard ${extraClassName ? extraClassName : ''}`}
    data-clipboard-action="copy"
    data-clipboard-text={value}
    disabled={disabled}
    type="button"
  />
)
