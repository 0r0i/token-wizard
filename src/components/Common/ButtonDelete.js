import React from 'react'

export const ButtonDelete = ({ onClick, extraClassName = '', disabled }) => (
  <button className={`sw-ButtonDelete ${extraClassName}`} disabled={disabled} onClick={onClick} type="button" />
)
