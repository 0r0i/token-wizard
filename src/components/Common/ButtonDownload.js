import React, { Component } from 'react'

export class ButtonDownload extends Component {
  render() {
    const { disabled, onClick, extraClass } = this.props
    return (
      <button
        className={`sw-ButtonDownload ${extraClass ? extraClass : ''}`}
        disabled={disabled}
        onClick={onClick}
        type="button"
      >
        Download
      </button>
    )
  }
}
