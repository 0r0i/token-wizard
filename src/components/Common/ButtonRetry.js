import React from 'react'

export const ButtonRetry = ({ disabled = false, onClick, extraClassName = '' }) => (
  <button className={`sw-ButtonRetry ${extraClassName}`} disabled={disabled} onClick={onClick} type="button">
    <span className={`sw-ButtonRetry_Text`}>Retry transaction</span>
  </button>
)
