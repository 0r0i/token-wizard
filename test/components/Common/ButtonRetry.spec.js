import React from 'react'
import render from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-15'
import { configure, shallow } from 'enzyme'
import { ButtonRetry } from '../../../src/components/Common/ButtonRetry'

configure({ adapter: new Adapter() })

describe(`ButtonRetry`, () => {
  it(`should be a <button> HTML tag`, () => {
    const wrapper = shallow(<ButtonRetry />)

    expect(wrapper.type()).toBe('button')
  })

  it(`its type attribute should be 'button'`, () => {
    const wrapper = shallow(<ButtonRetry />)

    expect(wrapper.props().type).toBe('button')
  })

  it(`should contain a span with the .sw-ButtonRetry_Text class`, () => {
    const theButtonText = 'The button text'
    const wrapper = shallow(<ButtonRetry buttonText={theButtonText} onClick={jest.fn()} />)
    const buttonChild = wrapper.find('span.sw-ButtonRetry_Text')

    expect(buttonChild.exists()).toBeTruthy()
  })

  it(`should render ButtonRetry component`, () => {
    // Given
    const component = render.create(<ButtonRetry onClick={jest.fn()} />)

    // When
    const tree = component.toJSON()

    // Then
    expect(tree).toMatchSnapshot()
  })
})
