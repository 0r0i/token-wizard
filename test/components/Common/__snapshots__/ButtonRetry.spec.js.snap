// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonRetry should render ButtonRetry component 1`] = `
<button
  className="sw-ButtonRetry "
  disabled={false}
  onClick={[MockFunction]}
  type="button"
>
  <span
    className="sw-ButtonRetry_Text"
  >
    Retry transaction
  </span>
</button>
`;
